flow/aggregates.py

```diff
@@ -521,10 +521,10 @@ def __len__(self):
         return len(self._project)
 
     def __eq__(self, other):
-        return type(self) == type(other) and self._project == other._project
+        return type(self) == type(other) and self._project_repr == other._project_repr
```

Comment:
Is it possible for the `repr`s to be equivalent but not the `type`s?

Reply:
If we dropped the type check, we would get an `AttributeError` (can't access `other._project_repr`) when comparing with another type, instead of the expected `False`.

However, the most correct way to implement `__eq__` (accounting for subclasses) is probably closer to what I implemented in https://github.com/glotzerlab/signac/pull/455. I'll adopt that here.

```suggestion
        if not isinstance(other, type(self)):
            return NotImplemented
        return self._project_repr == other._project_repr
```